### PR TITLE
Remove redundant casts

### PR DESCRIPTION
**Pull request by DanVanAtta**
Some more cleanup via static analysis tools - remove unnecessary casts

I checked rebasing on https://github.com/triplea-game/triplea/pull/743 for merge conflicts and things were reasonable. That makes sense since this PR removes casts we do not need outright, while #743 takes more effort to eliminate casts that were necessary.

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/757)

### CODE REVIEW

**Comment by ron-murhammer** [comment]
@DanVanAtta Looks like this needs rebased.

**Comment by DanVanAtta** [comment]
updated

File: src/games/strategy/triplea/ui/TripleAFrame.java [hunk 1]
```diff
@@ -1176,7 +1176,7 @@ public void run() {
           final int maxAllowed =
               Math.min(BattleDelegate.getMaxScrambleCount(possibleScramblers.get(from).getFirst()), possible.size());
           final UnitChooser chooser =
-              new UnitChooser(possible, Collections.emptyMap(), data, false, uiContext);
+              new UnitChooser(possible, Collections.<Unit, Collection<Unit>>emptyMap(), data, false, uiContext);
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/ui/TripleAFrame.java]: Bad merge?

File: src/games/strategy/triplea/ui/TripleAFrame.java [hunk 2]
```diff
@@ -1276,7 +1276,7 @@ public void run() {
         panelChooser.add(new JLabel(" "));
         final int maxAllowed = possible.size();
         final UnitChooser chooser =
-            new UnitChooser(possible, Collections.emptyMap(), data, false, uiContext);
+            new UnitChooser(possible, Collections.<Unit, Collection<Unit>>emptyMap(), data, false, uiContext);
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/ui/TripleAFrame.java]: Bad merge?

**Comment by DanVanAtta** [comment]
updated

**Pull request merged**